wallets/core/src/hub/hub.ts

```diff
@@ -0,0 +1,122 @@
+import type { Namespace, State as NamespaceState } from './namespaces/mod.js';
+import type { Provider, State as ProviderState } from './provider/mod.js';
+import type { Store } from './store/mod.js';
+
+type HubState = {
+  [key in string]: ProviderState & {
+    namespaces: NamespaceState[];
+  };
+};
+
+type RunAllResult = {
+  id: string;
+  provider: unknown;
+  namespaces: unknown[];
+};
+
+interface HubOptions {
+  store?: Store;
+}
+export class Hub {
+  #providers = new Map<string, Provider>();
+  #options: HubOptions;
+
+  constructor(options?: HubOptions) {
+    this.#options = options ?? {};
+    /*
+     * TODO:
+     * config:
+     * isEagerConnectEnabled? (warning if explicitly calls eagerConnect)
+     *
+     */
+    this.init();
+  }
+
+  init() {
+    this.runAll('init');
+  }
+
+  /*
+   * Running a specific action (e.g. init) on all namespaces and providers one by one.
+   *
+   *  TODO: Some of methods may accepts args, with this implementation we only limit to those one without any argument.
+   */
+  runAll(action: string): RunAllResult[] {
+    const output: RunAllResult[] = [];
+
+    // run action on all providers eagerConnect, disconnect
+    const providers = this.#providers.values();
+
+    for (const provider of providers) {
+      // Calling `action` on `Provider` if exists.
+      const providerOutput: RunAllResult = {
+        id: provider.id,
+        provider: undefined,
+        namespaces: [],
+      };
+
+      const providerMethod = provider[action as keyof Provider];
+      if (typeof providerMethod === 'function') {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore-next-line
+        providerOutput.provider = providerMethod.call(provider);
+      }
+
+      // Namespace instances can have their own `action` as well. we will call them as well.
+      const namespaces = provider.getAll().values();
+      for (const namespace of namespaces) {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore-next-line
+        const namespaceMethod = namespace[action];
+        if (typeof namespaceMethod === 'function') {
+          const result = namespaceMethod();
+          providerOutput.namespaces.push(result);
+        }
+      }
+
```

Comment:
On eager connect, we shouldn't call connect on all namespaces, we should call it on selected namespace by users. (previously connected namespaces)